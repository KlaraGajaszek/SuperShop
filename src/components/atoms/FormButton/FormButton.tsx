import React, { FC } from 'react';

type PropsButton = {
  content: string;
};

export const FormButton: FC<PropsButton> = ({ content }) => {
  return (
    <button
      data-testid="buttonID"
      type="submit"
      className="rounded-md bg-black text-yellow-100 font-semibold  focus:outline-none p-3 shadow-lg "
    >
      {content}
    </button>
  );
};
