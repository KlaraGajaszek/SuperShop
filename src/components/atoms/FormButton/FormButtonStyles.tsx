import styled from 'styled-components';

export const ButtonWrapper = styled.button<{
  bgcolor: string;
  radius: string;
  margin: string;
  width: string;
  color: string;
}>`
  border-radius: ${(props) => props.radius};
  border: none;
  outline: none;
  font-weight: bold;
  padding: 5px 10px;
  background-color: ${(props) => props.bgcolor};
  margin: ${(props) => props.margin};
  width: ${(props) => props.width};
  cursor: pointer;
  height: 30px;
  color: ${(props) => (props.color ? props.color : 'black')};
  :hover {
    background-color: ${(props) => props.color};
    color: ${(props) => props.bgcolor};
  }
`;
