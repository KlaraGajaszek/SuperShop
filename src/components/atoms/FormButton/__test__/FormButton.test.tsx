import React from 'react';
import 'jest-styled-components';
import { screen, render, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import { FormButton } from '../FormButton';

describe('Content', () => {
  it('render without crash', () => {
    render(<FormButton content="test" />);
    const formWrapper = screen.getByText('test');
    expect(formWrapper).toBeInTheDocument();
  });

  it('all styles should be render', () => {
    const formButton = renderer
      .create(<FormButton bgcolor="red" radius="5" margin="5" width="20" />)
      .toJSON();
    expect(formButton).toMatchSnapshot();
  });

  // it('function pass as onClick is called when user click on the button', () => {
  //   const mockCallback = jest.fn();
  //   render(<FormButton onClick={mockCallback} />);

  //   userEvent.click(screen.getByTestId('buttonID'));
  //   expect(mockCallback).toHaveBeenCalled();
  // });
});
