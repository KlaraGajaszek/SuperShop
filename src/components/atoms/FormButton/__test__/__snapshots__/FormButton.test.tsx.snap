// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content all styles should be render 1`] = `
<button
  className="rounded-md bg-black text-yellow-100 font-semibold  focus:outline-none p-3 shadow-lg "
  data-testid="buttonID"
  type="submit"
/>
`;
